src/function/gds/output_writer.cpp

```diff
@@ -19,32 +19,35 @@ void PathsOutputs::beginWritingOutputsForDstNodesInTable(common::table_id_t tabl
     bfsGraph.pinNodeTable(tableID);
 }
 
+static std::unique_ptr<ValueVector> createVector(const LogicalType& type,
```

Comment:
Can you put this into another common utility function? This doesn't look like a gds-related utility function. Looks like something many other parts of the system can use as well. Do we have a VectorUtils-like or CommonUtils-like class?

Reply:
I don't recall other parts using this. The part of assigning `SingleDataChunkState` is very specific to GDS.